app/core/controllers/errors_reporter_test.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for processing error reports."""
+
+from core.tests import app_engine_test_base
+
+import main
+
+
+class FrontendErrorHandlerTest(app_engine_test_base.AppEngineTestBase):
+    """Backend integration tests for handling frontend errors."""
+
+    def test_frontend_errors_are_logged(self):
+        """Test logging is done once a POST request is processed."""
+
+        errors_log = {
+            'error': 'scope errors'
+        }
+
+        self.testapp.post_json(main.FRONTEND_ERROR_URL, errors_log)
+        with self.assertLogs(None, level='INFO') as cm:
```

Comment:
I looked into backporting assertLogs from python3 instead (https://pypi.org/project/logging-test-case/) -- but actually I feel like this is overkill. Perhaps just do self.swap(logging, 'error', ...) instead and declare a custom function?

Reply:
so the file `logger_test.py` is the backported version of  [unittest's assertLogs](https://docs.python.org/3/library/unittest.html?highlight=unittest#unittest.TestCase.assertLogs), here is where I got it from.
https://stackoverflow.com/questions/20715042/python-2-7-unit-test-assert-logger-warning-thrown
I mean I can do a self.swap. I do feel like it is just cleaner to outright back-port it.

What makes you think it as overkill?

Reply:
self.swap would take about 10 lines max. Adding a whole new file for assertLogs takes much more than that, and it's harder to review and read (plus it comes with a whole bunch of stuff that's only applicable to more generalized scenarios).

If we actually did want to backport assertLogs we should include it as a third-party library and not have it in the core codebase as code that we maintain.

Reply:
Got it! I now use a third party library to take care of reading logs.